pymc3/gp/gp.py

```diff
@@ -63,56 +60,64 @@ def GP(name, X, mean_func=None, cov_func=None,
         mean_func = Zero()
     else:
         if not isinstance(mean_func, Mean):
-            raise GPValueError('mean_func must be a subclass of Mean')
-    if cov_func is None:
-        raise GPValueError('A covariance function must be specified for GP')
+            raise ValueError('mean_func must be a subclass of Mean')
     if not isinstance(cov_func, Covariance):
-        raise GPValueError('cov_func must be a subclass of Covariance')
-
-    # NONCONJUGATE
-    if observed is None:
+        raise ValueError('cov_func must be a subclass of Covariance')
+
+    # use marginal form for GP
+    if observed is not None:
+        # no approximation indicated
+        if all(value is None for value in [approx, n_inducing, inducing_points]):
+            # provide only one of sigma, cov_func_noise
+            if sigma is not None and cov_func_noise is not None:
+                raise ValueError(('Must provide one of sigma or '
+                                  'cov_func_noise'))
+            if sigma is None and cov_func_noise is None:
+                raise ValueError(('Must provide a value or a prior '
+                                  'for the noise variance'))
+            if sigma is not None and cov_func_noise is None:
+                cov_func_noise = WhiteNoise(X.shape[0], sigma)
+            return GPFullConjugate(name, X, mean_func, cov_func, cov_func_noise,
+                                   observed=observed)
+        else:
+            # use an approximation
+            if approx is None:
+                pm._log.info("Using VFE approximation")
+                approx = "VFE"
+
+            approx = approx.upper()
+            if approx not in ["VFE", "FITC"]:
+                raise ValueError(("'FITC' or 'VFE' are the implemented "
```

Comment:
better message should say that `'FITC' or 'VFE'` are the only approximations that are implemented and show what has user provided. e.g. `"'FITC' or 'VFE' are the only supported approximations for GP, got %s"`